i18n4k-core/src/commonMain/kotlin/de/comahe/i18n4k/LocaleUtils.kt

```diff
@@ -26,10 +31,36 @@ val Locale.country: String
 val Locale.variant: String
     get() = this.getVariant()
 
+/** Cache for [lessSpecificLocale] to prevent memory allocations. */
+private val lessSpecificLocaleCache = atomic(persistentMapOf<Locale, Locale?>())
+
 /**
- * Transforms a languageTag like "en_US_WIN" to a
- * Locale("en","US","WIN")
+ * Creates a locale that is one step less specific than this locale.
+ *
+ * Returns null if there is no less specific locale.
+ *
+ * E.g. "de_DE_saxony" would become "de_DE", "de_DE" would become "de" and
+ * "de" would become null.
  */
+val Locale.lessSpecificLocale: Locale?
```

Comment:
Sorry, finally circling around to this but I wonder if this is more clear as a `higherOrderLocale`. Just a thought.